storageClient/storageClient_test.go

```diff
@@ -330,3 +333,186 @@ func TestReadGroupWithValidInputSucceeds(t *testing.T) {
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	group = CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	response, err = queenClient.CreateGroup(testCtx, group)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	// List The Groups Under the Account
+	listRequest := ListGroupsRequest{}
+	responseList, err := queenClient.ListGroups(testCtx, listRequest)
+	if err != nil {
+		t.Fatalf("Failed to list Group: Response( %+v) \n error %+v", responseList, err)
+	}
+	t.Log(responseList)
+
+}
+
+func TestListGroupsWithClientForClientWithGroupsReturnsSuccess(t *testing.T) {
+	registrationClient := accountClient.New(e3dbClients.ClientConfig{Host: cyclopsServiceHost})
+	queenClientInfo, createAccountResponse, err := e3dbTest.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), cyclopsServiceHost)
+	if err != nil {
+		t.Fatalf("Error %s making new account", err)
+	}
+	queenClientInfo.Host = cyclopsServiceHost
+	accountToken := createAccountResponse.AccountServiceToken
+	//queenClient := storageClient.New(queenClientInfo)
+	queenAccountClient := accountClient.New(queenClientInfo)
+	registrationToken, err := e3dbTest.CreateRegistrationToken(&queenAccountClient, accountToken)
+	if err != nil {
+		t.Fatalf("error %s creating account registration token using %+v %+v", err, queenAccountClient, accountToken)
+	}
+
+	reg, ClientConfig, err := e3dbTest.RegisterClient(testCtx, ClientServiceHost, registrationToken, "name")
+	if err != nil {
+		t.Fatalf("Error registering Client %+v %+v %+v ", reg, err, ClientConfig)
+	}
+	//ClientConfig.AuthNHost = cyclopsServiceHost
+	ClientConfig.Host = cyclopsServiceHost
+	regClient := New(ClientConfig)
+
+	//Create A group
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	group := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	response, err := regClient.CreateGroup(testCtx, group)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+
+	listRequest := ListGroupsRequest{}
+	responseList, err := regClient.ListGroups(testCtx, listRequest)
+
+	if err != nil {
+		t.Fatalf("Group LIST: Group was Not Found: %+v ", responseList)
+	}
```

Comment:
this test is almost complete, suggest also checking that the response contains a group with the same id as returned on line 503

Reply:
comment above should not be marked as resolved